### PR TITLE
Render joint names

### PR DESCRIPTION
**Pull request by amzn-rhhong**
image

Signed-off-by: rhhong <rhhong@amazon.com>

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 1]
```diff
@@ -49,12 +52,23 @@ namespace AZ::Render
...
         }
 
-        // Render skeleton
+        // Render simple line skeleton
         if (renderFlags[EMotionFX::ActorRenderFlag::RENDER_LINESKELETON])
+        {
+            RenderLineSkeleton(instance, renderActorSettings.m_lineSkeletonColor);
+        }
+
+        // Render advance skeleton
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:61]: Nit: advance -> advanced

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 2]
```diff
@@ -189,6 +226,42 @@ namespace AZ::Render
         auxGeom->DrawLines(lineArgs);
     }
 
+    void AtomActorDebugDraw::RenderSkeleton(EMotionFX::ActorInstance* instance, const AZ::Color& skeletonColor)
+    {
+        RPI::AuxGeomDrawPtr auxGeom = m_auxGeomFeatureProcessor->GetDrawQueue();
+
+        const EMotionFX::TransformData* transformData = instance->GetTransformData();
+        const EMotionFX::Skeleton* skeleton = instance->GetActor()->GetSkeleton();
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:234]: Wasn't this part of another PR already?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 3]
```diff
@@ -487,4 +560,54 @@ namespace AZ::Render
...
+        }
+
+        const EMotionFX::Actor* actor = actorInstance->GetActor();
+        const EMotionFX::Skeleton* skeleton = actor->GetSkeleton();
+        const EMotionFX::TransformData* transformData = actorInstance->GetTransformData();
+        const EMotionFX::Pose* pose = transformData->GetCurrentPose();
+        const size_t numEnabledNodes = actorInstance->GetNumEnabledNodes();
+
+        m_drawParams.m_drawViewportId = viewportContext->GetId();
+        auto viewportSize = viewportContext->GetViewportSize();
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:590]: What is this type? Just a vec2? auto overuse?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 4]
```diff
@@ -487,4 +560,54 @@ namespace AZ::Render
...
+        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), 0.0f, 1.0f) +
+            TopRightBorderPadding * viewportContext->GetDpiScalingFactor();
+        m_drawParams.m_color = jointNameColor;
+        m_drawParams.m_scale = AZ::Vector2(BaseFontSize);
+        m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Right;
+        m_drawParams.m_monospace = false;
+        m_drawParams.m_depthTest = false;
+        m_drawParams.m_virtual800x600ScreenSize = false;
+        m_drawParams.m_scaleWithWindow = false;
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:599]: Do we need to set all of these? Some look like default values?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/RenderPlugin/RenderOptions.cpp [hunk 5]
```diff
@@ -1082,6 +1083,7 @@ namespace EMStudio
         settings.m_selectedClothColliderColor = m_selectedClothColliderColor;
         settings.m_simulatedObjectColliderColor = m_simulatedObjectColliderColor;
         settings.m_selectedSimulatedObjectColliderColor = m_selectedSimulatedObjectColliderColor;
+        settings.m_nodeNameColor = m_nodeNameColor;
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/RenderPlugin/RenderOptions.cpp:1086]: Nit: jointNameColor

**Review by amzn-jillich — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp (hunk 4 above)
Comment (amzn-rhhong) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:599]: Not really but we also don't want to depend on the default value

**Pull request merged**